src/redraw.mm

```diff
@@ -70,123 +70,65 @@ - (void) doAction:(RedrawCode::Enum)code withItem:(const object &)item_o
-            [textAttrs setDictionary:mTextAttrs];
-            NSColor *fg = [textAttrs objectForKey:NSForegroundColorAttributeName];
-            NSColor *bg = [textAttrs objectForKey:NSBackgroundColorAttributeName];
-            [textAttrs setObject:fg forKey:NSBackgroundColorAttributeName];
-            [textAttrs setObject:bg forKey:NSForegroundColorAttributeName];
-        }
-        else {
-            textAttrs = mTextAttrs;
+            nsrun = [nsrun stringByAppendingString:
+              [NSString stringWithUTF8String:char_s.c_str()]];
         }
 
-        for (int i=0; i<runs.size(); i++) {
-            const std::string &run = runs[i];
-            int sz = lens[i];
+        // Force left-to-right rendering
+        nsrun = [nsrun stringByAppendingString:@"\u202c"];
 
-            if (sz == 0)
-                continue;
-
-            NSString *nsrun = [NSString stringWithUTF8String:run.c_str()];
+        CGContextSaveGState(mCanvasContext);
+        // TODO: Remove when font height formula is corrected (updateCharSize)
+        CGContextClipToRect(mCanvasContext, bgrect);
 
-            // Force left-to-right rendering
-            nsrun = [@"\u202d" stringByAppendingString:nsrun];
-            nsrun = [nsrun stringByAppendingString:@"\u202c"];
+        NSPoint point = [self
+            viewPointFromCellPoint:CGPointMake(mCursorPos.x, mCursorPos.y)];
 
-            NSRect cellRect = CGRectMake(mCursorPos.x, mCursorPos.y, sz, 1);
-            NSRect rect = [self viewRectFromCellRect:cellRect];
+        [nsrun drawAtPoint:point withAttributes:textAttrs];
 
-            /* Maybe there is some combination of options for either drawAtPoint,
-            drawInRect, or drawWithRect, that makes Cocoa draw some text in a
-            fucking rectangle, but I couldn't figure it out. The background is
-            always too tall or too short. Solution:
-
-            - Draw our own background for fonts like Monaco that come up short
-            - Use a clipping rect for fonts like Droid Sans that draw way too
-                high */
-
-            NSColor *bg = [textAttrs objectForKey:NSBackgroundColorAttributeName];
-            [bg set];
-            NSRectFill(rect);
-
-            CGPoint origin = rect.origin;
-            float r = rect.origin.x + rect.size.width;
-            rect.origin.x = floor(rect.origin.x);
-            rect.size.width = ceil(r - rect.origin.x);
-
-            CGContextSaveGState(mCanvasContext);
-            CGContextClipToRect(mCanvasContext, rect);
-            [nsrun drawAtPoint:origin withAttributes:textAttrs];
-            CGContextRestoreGState(mCanvasContext);
-
-            mCursorPos.x += sz;
-        }
+        mCursorPos.x += item_sz - 1;
 
+        CGContextRestoreGState(mCanvasContext);
         [self setNeedsDisplay:YES];
     }
-    else for (int i=0; i<narglists; i++) {
+    else for (int i = 0; i < item_sz - 1; i++) {
```

Comment:
Yea, I removed it as it was just used at that place. I thought it's pretty obvious. We can still add a comment about this, but I guess it's not a big deal.